pc/protocols/usb_boot.c

```diff
@@ -332,9 +331,8 @@ usbBootError_t usb_find_device_with_bcd(unsigned idx, char *input_addr,
                     *device = dev;
                     devs = 0;
 
-                    int speed = libusb_get_device_speed(dev);
+                    usb_speed_enum = libusb_get_device_speed(dev);
```

Comment:
If you are 100% sure that libusb will keep the same enumeration, then below isn't necessary, otherwise rather do a switch.
```
switch (libusb_get_device_speed(dev))
{
case LIBUSB_SPEED_UNKNOWN: 
    usb_speed_enum = XLINK_USB_SPEED_UNKNOWN;
    break;
case LIBUSB_SPEED_LOW: 
    usb_speed_enum = XLINK_USB_SPEED_LOW;
    break;
case LIBUSB_SPEED_FULL: 
    usb_speed_enum = XLINK_USB_SPEED_FULL;
    break;
case LIBUSB_SPEED_HIGH: 
    usb_speed_enum = XLINK_USB_SPEED_HIGH;
    break;
case LIBUSB_SPEED_SUPER: 
    usb_speed_enum = XLINK_USB_SPEED_SUPER;
    break;
case LIBUSB_SPEED_SUPER_PLUS:
    usb_speed_enum = XLINK_USB_SPEED_SUPER_PLUS;
break; 
default:
    usb_speed_enum = XLINK_USB_SPEED_UNKNOWN;
    break;
}
```

Reply:
I am using the same enumeration so I didn't so that part.

Reply:
Do I need to use it just to be on the safer side ?

Reply:
From one perspective yes, but libusb being a library as stable as it gets, they most likely won't change their speed enumeration sequence, in which case a cast/directly assigning the value will do. In that case you can mark the UsbSpeed_t enumerations with same values as libusb to be more explicit:
Libusb enumeration:
```
libusb_speed {
  LIBUSB_SPEED_UNKNOWN = 0, LIBUSB_SPEED_LOW = 1, LIBUSB_SPEED_FULL = 2, LIBUSB_SPEED_HIGH = 3,
  LIBUSB_SPEED_SUPER = 4, LIBUSB_SPEED_SUPER_PLUS = 5
}
```
UsbSpeed_t enumeration:
```
typedef enum {
    XLINK_USB_SPEED_UNKNOWN = 0,
    XLINK_USB_SPEED_LOW = 1,
    XLINK_USB_SPEED_FULL = 2,
    XLINK_USB_SPEED_HIGH = 3,
    XLINK_USB_SPEED_SUPER = 4,
    XLINK_USB_SPEED_SUPER_PLUS = 5
} UsbSpeed_t;
```

Reply:
got it. makes sense. I don't see any  `LIBUSB_SPEED_SUPER_PLUS` in their docs. and I was thinking of using this `X_LINK_USB_SPEED_SUPER` format since every enum I saw in XLinkPublicDefines.h is defined that way.

Reply:
You are probably browsing some university mirror of outdated docs, I also didn't see the updated enum in there. Here are the (probably) latest: http://libusb.sourceforge.net/api-1.0/group__libusb__dev.html
Yes, use the latter. The former was just from libusb spec, to show that direct conversion will work as long as the enumeration is 1:1 (eg: LIBUSB_SPEED_LOW = 1 == XLINK_USB_SPEED_LOW = 1, ...)

Reply:
got it. Thanks.